implementations/rust/ockam/ockam_command/src/identity/delete.rs

```diff
@@ -0,0 +1,37 @@
+use crate::util::{exitcode, node_rpc};
+use crate::CommandGlobalOpts;
+use clap::Args;
+use anyhow::anyhow;
+use ockam::Context;
+
+#[derive(Clone, Debug, Args)]
+pub struct DeleteCommand {
+    name: String,
+}
+
+impl DeleteCommand {
+    pub fn run(self, options: CommandGlobalOpts) {
+        node_rpc(run_impl, (options, self))
+    }
+}
+
+async fn run_impl(
+    ctx: Context,
+    (opts, cmd): (CommandGlobalOpts, DeleteCommand),
+) -> crate::Result<()> {
+    let identity_state = opts.state.identities.get(&cmd.name)?;
+    let identifier = identity_state.config.identifier.to_string();
+    for node in opts.state.nodes.list()? {
+        let node_identity = node.config.identity(&ctx).await?;
+        let node_identifier = node_identity.identifier().to_string();
+        if node_identifier == identifier {
```

Comment:
Similar to what I proposed in the previous comment, maybe it would make more sense to compare the identifiers objects directly:

```rust
let identity = opts.state.identities.get(&cmd.name)?;
for node in opts.state.nodes.list()? {
    let node_identity = node.config.identity(&ctx).await?;
    if node_identity.identifier() == identity.config.identifier {
    // ...
```

Reply:
Thanks for the feedback, changes made!